src/core/quantum_gate.jl

```diff
@@ -93,21 +99,143 @@ q[2]:───────
 ```
 
 """
-abstract type AbstractGate end
+abstract type AbstractGateSymbol end
+
+abstract type AbstractControlledGateSymbol <: AbstractGateSymbol end
+
+# TODO(#226): delete on completion
+get_connected_qubits(gate::AbstractGateSymbol)=[gate.target]
+
+get_gate_parameters(gate::AbstractGateSymbol)=Dict{String,Real}()
+
+get_num_connected_qubits(gate::AbstractGateSymbol)=length(get_connected_qubits(gate))
+
+# TODO(#226): `Gate` should not inherit from `AbstractGateSymbol` after completion
+"""
+    Gate
+
+An object that specifies an `AbstractGateSymbol` and its placement inside a `QuantumCircuit`. The placement is the same as the target qubits on which the gate operates.
+
+# Examples
+```jldoctest
+julia> gate = Gate(iswap(1, 2), [1, 2])
+Gate Object: Snowflake.ISwap
+Connected_qubits	: [1, 2]
+Operator:
+(4, 4)-element Snowflake.SwapLikeOperator:
+Underlying data ComplexF64:
+Equivalent DenseOperator:
+1.0 + 0.0im    0.0 + 0.0im    0.0 + 0.0im    0.0 + 0.0im
+0.0 + 0.0im    0.0 + 0.0im    0.0 + 1.0im    0.0 + 0.0im
+0.0 + 0.0im    0.0 + 1.0im    0.0 + 0.0im    0.0 + 0.0im
+0.0 + 0.0im    0.0 + 0.0im    0.0 + 0.0im    1.0 + 0.0im
+```
+
+
+```jldoctest
+julia> gate = Gate(universal(3, pi/2, -pi/2, pi/2), [3])
+Gate Object: Snowflake.Universal
+Parameters:
+theta	: 1.5707963267948966
+phi	: -1.5707963267948966
+lambda	: 1.5707963267948966
+
+Connected_qubits	: [3]
+Operator:
+(2, 2)-element Snowflake.DenseOperator:
+Underlying data ComplexF64:
+0.7071067811865476 + 0.0im    -4.329780281177466e-17 - 0.7071067811865475im
+4.329780281177466e-17 - 0.7071067811865475im    0.7071067811865476 + 0.0im
+```
+"""
+struct Gate{GateType} <: AbstractGateSymbol where GateType <: AbstractGateSymbol
+    gate::GateType
```

Comment:
changing the `gate` field to `gate_symbol` would be less ambiguous. Then `get_gate(gate::Gate)` could be `get_gate_symbol(gate::Gate)